packages/editor/src/bindings/pattern-overrides.js

```diff
@@ -2,10 +2,82 @@
  * WordPress dependencies
  */
 import { _x } from '@wordpress/i18n';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+
+const CONTENT = 'content';
 
 export default {
 	name: 'core/pattern-overrides',
 	label: _x( 'Pattern Overrides', 'block bindings source' ),
-	useSource: null,
+	getValue( { registry, clientId, attributeName } ) {
+		const { getBlockAttributes, getBlockParents, getBlockName } =
+			registry.select( blockEditorStore );
+		const currentBlockAttributes = getBlockAttributes( clientId );
+		const parents = getBlockParents( clientId, true );
+		const patternClientId = parents.find(
+			( id ) => getBlockName( id ) === 'core/block'
+		);
+
+		const overridableValue =
+			getBlockAttributes( patternClientId )?.[ CONTENT ]?.[
+				currentBlockAttributes?.metadata?.name
+			]?.[ attributeName ];
+
+		// If there is no pattern client ID, or it is not overwritten, return the default value.
+		if ( ! patternClientId || overridableValue === undefined ) {
+			return currentBlockAttributes[ attributeName ];
+		}
+
+		return overridableValue === '' ? undefined : overridableValue;
+	},
+	setValues( { registry, clientId, attributes } ) {
+		const { getBlockAttributes, getBlockParents, getBlockName, getBlocks } =
+			registry.select( blockEditorStore );
+		const currentBlockAttributes = getBlockAttributes( clientId );
+		const parents = getBlockParents( clientId, true );
+		const patternClientId = parents.find(
+			( id ) => getBlockName( id ) === 'core/block'
+		);
```

Comment:
Nit: We can just use `getBlockParentsByBlockName`.

```suggestion
		const patternClientId = getBlockParentsByBlockName( clientId, 'core/block', true )[0];
```

Reply:
I changed the logic to use it in [this commit](https://github.com/WordPress/gutenberg/pull/60721/commits/10fa4e8df34e2c5ba8b719227f5f0a3c9959c950). Thanks for the tip! I didn't know about that function.